safe/impact_functions/inundation/flood_polygon_roads/impact_function.py

```diff
@@ -71,12 +71,12 @@ def run(self, layers=None):
         self.prepare(layers)
 
         # Set the target field
-        target_field = 'FLOODED'
+        target_field = self.parameters['target_field_value'].value
 
         # Get the parameters from IF options
-        road_type_field = self.parameters['road_type_field']
-        affected_field = self.parameters['affected_field']
-        affected_value = self.parameters['affected_value']
+        road_type_field = self.parameters['road_type_field'].value
```

Comment:
As above we want to move this logic to more like this:

```
self.hazard_metadata.type_field
self.exposure_metadata.affected_field
self.exposure_metadata.affected_value
```

Maybe we must merge this as you have it for now and do the above in a second pass, but please liaise with @akbargumbira and @ismailsunni to sequence this work.